### DISCUSSION TITLE
Memif/vhost-user DPDK Drivers

### DISCUSSION THREAD

**Discussion by infinitydon** [srsran/srsRAN_Project#461] [General Help]
Hi,

Does the DPDK feature of the CU/DU support other implementations like memif/vhost-user apart from vfio-pci..

Sample architecture is shown below:

          +-----------+
          |    CU/DU    |
          memif0/1-slave
          +-----------+
               || 
       +----------------+
          memif0/1-master
          |    VPP    |
         memif0/2-master
       +-----------------+
              ||
          +-----------+
         memif0/2-slave
          |    UPF    |
          +-----------+

Example reasons for this kind of config:
- Situations where single physical interface is to be shared among the CNFs, whereby a central VPP/OVS-DPDK will be hosting the physical interface and then other CNFs will connect to the central VPP/OVS-DPDK via memif/vhost-user
- Service chaining within the same node while ensuring high throughput

**Comment by andrepuschmann** [comment]
Hey @infinitydon,
we've not used the memif PMD ourselves as there is no use-case just yet. Right now DPDK is only used for the front-haul. If we need to share a physical NIC we create VFs and to each of the VFs.

**Reply by infinitydon** [reply]
Ok. One other aspect is the case of public cloud VMs where PFs are not available and usually there is a limitation in the number of NICs that can be attached to a VM, in this case memif will be beneficial.

Just thought of sharing in case you consider this in the future.
